lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1390,7 +1390,8 @@ createValueConstructor(ClangImporter::Implementation &Impl,
         continue;
 
       if (auto clangField = dyn_cast<clang::FieldDecl>(var->getClangDecl()))
-        if (clangField->isAnonymousStructOrUnion())
+        if (clangField->isAnonymousStructOrUnion() ||
+            clangField->getDeclName().isEmpty())
```

Comment:
This change *probably* doesn't get exercised by the test. You could *probably* fix that by using `T(S(uc: 0))` instead of `T()`.

(This is simply from reading the code, though, so I could be missing a subtlety or two.)

Also see my note on the third use about simplifying the check.